cmd/update.go

```diff
@@ -98,6 +99,10 @@ func update(dryRun bool) error {
 	latest, found, err := updater.DetectLatest(slug)
 
 	if err != nil {
+		if strings.Contains(err.Error(), "401 Bad credentials") {
```

Comment:
That's a very good point, I did a similar crawl from the `selfupdate` library and wound up with the same conclusion.

Reply:
That would also lead me to believe this code is never executed :thinking: 
https://github.com/rhysd/go-github-selfupdate/blob/41c1bbb0804a2994dae69502a8c76e7c456ad45e/selfupdate/detect.go#L123

Reply:
This is definitely executed, since I was seeing that 401 error previously and on this branch I see this:

```
 $ go run main.go update check
Error: Your Github token is invalid. Check the [github] section in ~/.gitconfig
: GET https://api.github.com/repos/CircleCI-Public/circleci-cli/releases: 401 Bad credentials []
exit status 255
```

Reply:
Thanks for the link @zzak , `selfupdate` swallows 404 errors only and does an early exit with other kinds of errors. I guess Github returns a 401 if you pass a bad token. I've been testing this by truncating my user token, as no-token works just fine.

https://github.com/rhysd/go-github-selfupdate/blob/41c1bbb0804a2994dae69502a8c76e7c456ad45e/selfupdate/detect.go#L124

(Still not sure how my `.gitconfig` got into a state that led me to this error. Possibly a vim fat finger from me, but I don't recall editing this file...).

Reply:
Ah, go-github actually wraps `net/http` but provides the same interface, so I didn't notice it the first time I looked. It [handles http error codes](https://github.com/google/go-github/blob/master/github/github.go#L689-L727) and returns this: https://github.com/google/go-github/blob/master/github/github.go#L556-L576

I think the error could be checked by using the type:

```go

if errResponse, ok := err.(*github.ErrorResponse); ok && errResponse.Response.StatusCode == http.StatusUnauthorized {
    ...
}

```

Reply:
It looks like the error comes from the Github library's `CheckResponse`, which _does_ throw an error when the server returns non-2xx status codes.

https://github.com/CircleCI-Public/circleci-cli/blob/CIRCLE-13279-improve-github-auth-error/vendor/github.com/google/go-github/github/github.go#L682

Reply:
I like that suggestion! I'll update the PR with it. (Also I'm just seeing now that I didn't refresh this page before looking for the source) 🤦‍♂️